### DISCUSSION TITLE
Viwer don't work due to psutil stale

### DISCUSSION THREAD

**Discussion by Twanaar** [swar/Swar-Chia-Plot-Manager#304] [Q&A]
Hello, 

Manager and viewer are working since 2-3 days, but now the viewer have a problem. The manager still working:
![image](https://user-images.githubusercontent.com/17991616/117540381-1fbc4f00-afdd-11eb-8ec6-9dfb678d1753.png)

**Comment by Damme** [comment]
Same issue here, Linux. Just pulled the git.

**Comment by Ocidentally** [comment]
Yes, getting the same message myself

**Comment by Ocidentally** [comment]
I just closed the viewer and reopened in PowerShell as Administrator and started it again, and this seems to make it work again.

**Comment by Twanaar** [comment]
not working for me

**Comment by GreatPhoenix** [comment]
Mine is also having the same problem ever since I updated. After running in PowerShell as Admin. I have also tried running the program in a virtual python environment like in the readme but I am still getting this same error

**Comment by Twanaar** [comment]
after rebooting my computer, all working fine again.

Le dim. 9 mai 2021 à 20:43, GreatPhoenix ***@***.***> a
écrit :

**Comment by swar** [comment]
This typically ends up being an issue with Windows permissions. I haven't had this issue happen to me or been able to recreate it which is why I added the psutil stale fix that was a temporary fix. If it keeps happening the only advice I can give is to restart your PC. However, this issue does not stop manager from running, so manager will continue to run in the background.

**Comment by mccxiv** [comment]
In my case I had forgotten to change this path from the default (I don't have an `S:` drive)
```
log:
  # folder_path: This is the folder where your log files for plots will be saved.
  folder_path: S:\Chia\Logs\Plotter
```

Once I changed to a valid path and restarted the manager, it worked fine

**Comment by mrjackson** [comment]
This is the full error that I get. It only happens on Windows10, Server 2019 has been fine, as well as linux. The only fix I've found is restarting but it comes back after a couple days.

```Restarting view due to psutil going stale...
Traceback (most recent call last):
  File "C:\bin\Swar-Chia-Plot-Manager\manager.py", line 36, in <module>
    view()
  File "C:\bin\Swar-Chia-Plot-Manager\plotmanager\library\utilities\commands.py", line 123, in view
    os.execv(sys.executable, system_args)
OSError: [Errno 12] Not enough space```

**Comment by GreatPhoenix** [comment]
Yep i finally fixed mine by doing a full restart and now the viewer is working. Also even if this error exists your computer will still create plots you just don't get stats on it.

**Comment by Kali123411** [comment]
First time running it. Viewer crashed on me and did not restart but it seems that the manager is still chugging along....viewer says psutil went stale but did not restart just frozen. I'm on windows.

**Comment by Kali123411** [comment]
I restarted the viewer in Powershell like said above and it worked for only one second then gave me the psutil stale, it showed I had 3 plots in phase 1 but they all show zero progress. My CPU is pegged out at 98% and my SSD is filling up with plots though.

**Comment by Kali123411** [comment]
> This typically ends up being an issue with Windows permissions. I haven't had this issue happen to me or been able to recreate it which is why I added the psutil stale fix that was a temporary fix. If it keeps happening the only advice I can give is to restart your PC. However, this issue does not stop manager from running, so manager will continue to run in the background.

What permissions need to be given to the app?

**Comment by Kali123411** [comment]
> This is the full error that I get. It only happens on Windows10, Server 2019 has been fine, as well as linux. The only fix I've found is restarting but it comes back after a couple days.
> 
> ```
> Traceback (most recent call last):
>   File "C:\bin\Swar-Chia-Plot-Manager\manager.py", line 36, in <module>
>     view()
>   File "C:\bin\Swar-Chia-Plot-Manager\plotmanager\library\utilities\commands.py", line 123, in view
>     os.execv(sys.executable, system_args)
> OSError: [Errno 12] Not enough space```
> ```

If I restart my pc will I lose those plots and have to start over again?

**Comment by Kali123411** [comment]
> Hello,
> 
> Manager and viewer are working since 2-3 days, but now the viewer have a problem. The manager still working:
> ![image](https://user-images.githubusercontent.com/17991616/117540381-1fbc4f00-afdd-11eb-8ec6-9dfb678d1753.png)

Does the progress ever update for you ?

**Comment by mrjackson** [comment]
> If I restart my pc will I lose those plots and have to start over again?

Yes, the ones in progress will stop.

> Does the progress ever update for you?

Mine will look like this also. The phase will update but that's it. The manager does keep going properly, just the viewer acts up.

The only temporary fix for the viewer is to restart.

**Comment by Kali123411** [comment]
I rebooted my machine and now am running the viewer in PowerShell admin and it has been working since last night. I made some adjustments to my config so now my CPU is not maxing out anymore, seems to be running better.
